### ISSUE TITLE
[BUG] Getting a cryptic error message when calling getManifest()

### ISSUE THREAD

**Issue by nicholas-roether** [Hexer10/youtube_explode_dart#80]
**Describe the bug**
In my project, while simply trying to get the manifest for any Youtube video,
I often get strange and cryptic error messages, specifically when calling `streamsClient.getManifest()`:

```
NoSuchMethodError (NoSuchMethodError: The getter 'js' was called on null.
```

```
FormatException (FormatException: Unexpected character (at character 8156)
...er","width":"100%","height":"100%"}}});yt.setConfig({GAPI_HINT_PARAMS: "...
                                       ^
)
```

They seem to occur randomly as sometimes it works just fine, especially when loading a video that hasn't been loaded before (independent of application restarts), but I can't find a pattern in the crashes for the life of me.

The errors persist between restarts of the app, the VM, my PC and when switching between my PC and my laptop.

**To Reproduce**
As I already said, this code only sometimes produces the described issue.
```dart
var yt = YoutubeExplode();
yt.videos.streamsClient.getManifest("-FlxM_0S2lA");
```

**Stacktrace**
First error:
```
[ERROR:flutter/lib/ui/ui_dart_state.cc(177)] Unhandled Exception: NoSuchMethodError: The getter 'js' was called on null.
Receiver: null
Tried calling: js
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:51:5)
#1      _PlayerConfig.sourceUrl
#2      StreamsClient._getStreamContextFromWatchPage
<asynchronous suspension>
#3      StreamsClient.getManifest
...
```

Second error:
```
[ERROR:flutter/lib/ui/ui_dart_state.cc(177)] Unhandled Exception: FormatException: Unexpected character (at 
...er","width":"100%","height":"100%"}}});yt.setConfig({GAPI_HINT_PARAMS: "...
                                       ^

#0      _ChunkedJsonParser.fail (dart:convert-patch/convert_patch.dart:1404:5)
#1      _ChunkedJsonParser.parse (dart:convert-patch/convert_patch.dart:928:13)
#2      _parseJson (dart:convert-patch/convert_patch.dart:40:10)
#3      JsonDecoder.convert (dart:convert/json.dart:505:36)
#4      JsonCodec.decode (dart:convert/json.dart:156:41)
#5      EmbedPage.playerconfig
#6      StreamsClient._getStreamContextFromVideoInfo
<asynchronous suspension>
#7      StreamsClient.getManifest
...
```

**Enviroment**
 - Enviroment: Flutter
 - Version 1.22.2

**Additional context**
I don't have much experience with this, but for me this seems like there is a discrepancy between the data youtube is sending and what the library is expecting. I have no way of knowing, but perhaps youtube recently changed something about the way they transmit video information, so that it only sometimes conforms to the format that youtube_explode_dart assumes?

Even if this issue isn't reproducable, I would very much appreciate some ideas as to what the problem might be and how i might fix it; I've been wrecking by brain about this to no avail for quite some time now.

**Comment by nicholas-roether** [comment]
For now I temporarily solved this issue by repeatedly trying to get the manifest until it works. I hope that there is a better way to fix this though.

```
Future<T> retry<T>(Future<T> Function() futureBuilder, int times) async {
  var error;
  while (times > 0) {
    try {
      return await futureBuilder();
    } catch (e) {
      times--;
      error = e;
    }
  }
  throw error;
}

retry(() => yt.videos.streamsClient.getManifest("-FlxM_0S2lA"), 10);
```

**Comment by Hexer10** [comment]
This is already been fixed in YoutubeExplode, should be able to fix this shortly

**Comment by nicholas-roether** [comment]
Great, thanks a lot!

**Comment by Andrei-Portales** [comment]
This problem is already fixed?

**Comment by Andrei-Portales** [comment]
I have the same problem

I/flutter (13510): NoSuchMethodError: The getter 'js' was called on null.
I/flutter (13510): Receiver: null
I/flutter (13510): Tried calling: js

**Comment by nicholas-roether** [comment]
It seems they‘re working on it

**Comment by Hexer10** [comment]
I'm not at home right now, expect this to be fixed in a couple of hours

**Comment by dankydev** [comment]
Is this the same reason why getManifest always returns VideoUnavailableException lately? I saw they updated the original library and some fixes are due to Youtube's changes

**Comment by Hexer10** [comment]
Should be fixed in `1.6.2` please update! If this issue still persist, add a comment here.

> Is this the same reason why getManifest always returns VideoUnavailableException lately? I saw they updated the original library and some fixes are due to Youtube's changes

I'm not sure about this

**Comment by Abhishek0116** [comment]
Thanks to @nicholas-roether for exactly explain what is happening and also thanks to @Hexer10  for fixing the issue quickly.
